src/lib.rs

```diff
@@ -223,6 +231,13 @@ impl CertPaths {
 /// subject to the rules outlined above for `file`. The directory is not
 /// scanned recursively and may be empty.
 pub fn load_certs_from_paths(file: Option<&Path>, dir: Option<&Path>) -> CertificateResult {
+    load_certs_from_paths_internal(file, dir.map(|p| vec![p]).as_deref())
+}
+
+fn load_certs_from_paths_internal<P: AsRef<Path>>(
+    file: Option<&Path>,
+    dir: Option<&[P]>,
```

Comment:
I don't think `Option<&[P]>` makes sense, we can just pass in `&[P]` directly. Also, prefer `&[impl AsRef<Path>]` instead of the indirection via declaring `P`.